src/TSID/include/BipedalLocomotion/TSID/JointRegularizationTask.h

```diff
@@ -0,0 +1,98 @@
+    /**
+     * Initialize the planner.
+     * @param paramHandler pointer to the parameters handler.
+     * @param variablesHandler reference to a variables handler.
+     * @note the following parameters are required by the class
+     * |           Parameter Name           |   Type   |                                       Description                                      | Mandatory |
+     * |:----------------------------------:|:--------:|:--------------------------------------------------------------------------------------:|:---------:|
+     * | `robot_acceleration_variable_name` | `string` | Name of the variable contained in `VariablesHandler` describing the robot acceleration |    Yes    |
+     * |             `kp       `            | `vector` |                                Proportional controller gain                            |    Yes    |
+     * |             `kd`                   | `vector` |      Derivative Gain of the controller. if not specified \f$k_d = 2 \sqrt{k_p}\f$      |    No     |
+     * @return True in case of success, false otherwise.
+     */
+    bool initialize(std::weak_ptr<ParametersHandler::IParametersHandler> paramHandler,
+                    const System::VariablesHandler& variablesHandler) override;
+
+    /**
+     * Update the content of the element.
+     * @return True in case of success, false otherwise.
+     */
+    bool update() override;
+
+    /**
+     * Set the desired reference trajectory.
+     * @param jointPosition vector containing the desired joint position in radians.
+     * @note The desired velocity and acceleration are implicitly set to zero.
+     * @return True in case of success, false otherwise.
+     */
+    bool setReferenceTrajectory(Eigen::Ref<const Eigen::VectorXd> jointPosition);
+
+    /**
+     * Set the desired reference trajectory.
+     * @param jointPosition vector containing the desired joint position in radians.
+     * @param jointVelocity vector containing the desired joint velocity in radians per second.
+     * @note The desired acceleration is implicitly set to zero.
+     * @return True in case of success, false otherwise.
+     */
+    bool setReferenceTrajectory(Eigen::Ref<const Eigen::VectorXd> jointPosition,
+                                Eigen::Ref<const Eigen::VectorXd> jointVelocity);
+
+    /**
+     * Set the desired reference trajectory.
+     * @param jointPosition vector containing the desired joint position in radians.
+     * @param jointVelocity vector containing the desired joint velocity in radians per second.
+     * @param jointAcceleration vector containing the desired joint velocity in radians per second square.
+     * @return True in case of success, false otherwise.
+     */
+    bool setReferenceTrajectory(Eigen::Ref<const Eigen::VectorXd> jointPosition,
+                                Eigen::Ref<const Eigen::VectorXd> jointVelocity,
+                                Eigen::Ref<const Eigen::VectorXd> jointAcceleration);
```

Comment:
You are not setting trajectories, but rather desired set points.

Reply:
changed here: https://github.com/dic-iit/bipedal-locomotion-framework/pull/170/commits/40b888299f1a7925bac7d85b22fc7680ff88b922

Reply:
Just to mess up with you a little more :grin: ``setSetpoints``sounds funny, ``setReferencePoint``? :grin: